shared/src/main/scala/VNum.scala

```diff
@@ -46,14 +46,17 @@ object VNum:
     // Spits into real and imaginary parts
     // todo handle empty real part
     val parts = s.split("ı")
-    val real = parts(0)
-    val imag = parts.lift(1).getOrElse("0")
+    val real = parts.lift(0).getOrElse("0")
+    val imag = if s.endsWith("ı") then "1" else parts.lift(1).getOrElse("0")
 
-    val realNum = (if real.last == '.' then real + "5" else real).toInt
-    val imagNum = (if imag.last == '.' then imag + "5" else imag).toInt
+    // Apologies for not using val - I need to use the same variable twice
+    var realNum: String = if real.last == '.' then real + "5" else real
+    realNum = if realNum.startsWith(".") then "0" + realNum else realNum
+    var imagNum: String = if imag.last == '.' then imag + "5" else imag
+    imagNum = if imagNum.startsWith(".") then "0" + imagNum else imagNum
 
-    if imagNum == 0 then complex(realNum, 0)
-    else complex(realNum, imagNum)
+    if Real(imagNum) == 0 then complex(Real(realNum), 0)
```

Comment:
May I ask what this check is for? Isn't the 0 passed explicitly to `complex` going to be wrapped in a `Real` anyway?

Reply:
I don't know - it was around from before the PR, so I figured I needed to keep it.